sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1015 @@
+            checkContainsAny(MSG_RETURN_VALUE_WITH_NULLABLE,
+                    NullnessAnnotation.NULLABLE);
+            checkContainsAll(MSG_CONTRADICTING_RETURN_VALUE_ANNOTATIONS, NullnessAnnotation.NONNULL,
+                    NullnessAnnotation.CHECK_FOR_NULL);
+            checkContainsAll(MSG_OVERRIDDEN_METHOD_WITH_CHECK_RETURN_VALUE,
+                    NullnessAnnotation.CHECK_RETURN_VALUE, NullnessAnnotation.OVERRIDE);
+            checkRedundancyDueToClassLevelAnnotation(MSG_REDUNDANT_NONNULL_BY_DEFAULT_ANNOTATION,
+                    NullnessAnnotation.PARAMETERS_ARE_NONNULL_BY_DEFAULT);
+            checkRedundancyDueToClassLevelAnnotation(MSG_REDUNDANT_NULLABLE_BY_DEFAULT_ANNOTATION,
+                    NullnessAnnotation.PARAMETERS_ARE_NULLABLE_BY_DEFAULT);
+
+            if (isVoid()) {
+                checkContainsAny(MSG_VOID_WITH_CHECK_RETURN_VALUE_ANNOTATION,
+                        NullnessAnnotation.CHECK_RETURN_VALUE);
+            }
+            if (isPrimitiveType()) {
+                checkContainsAny(MSG_PRIMITIVES_WITH_NULLNESS_ANNOTATION,
+                        NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NONNULL,
+                        NullnessAnnotation.NULLABLE);
+            }
+            else {
+                final NullnessAnnotation annotation = getParentMethodOrClassAnnotation(
+                        NullnessAnnotation.RETURN_VALUES_ARE_NONNULL_BY_DEFAULT);
+                final boolean returnValuesAreNonnullByDefault = annotation
+                        == NullnessAnnotation.RETURN_VALUES_ARE_NONNULL_BY_DEFAULT;
+                final boolean isMethodOverridden = isMethodOverridden();
+
+                if (returnValuesAreNonnullByDefault) {
+                    if (!isMethodOverridden) {
+                        checkContainsAny(MSG_REDUNDANT_NONNULL_RETURN_ANNOTATION,
+                                NullnessAnnotation.NONNULL);
+                    }
+                }
+                else {
+                    checkContainsNone(MSG_RETURN_WITHOUT_NULLNESS_ANNOTATION,
+                            NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NONNULL,
+                            NullnessAnnotation.OVERRIDE);
+                }
+
+                if (isMethodOverridden && !allowOverridingReturnValue) {
+                    checkContainsAny(MSG_OVERRIDDEN_METHODS_ALLOW_ONLY_NONNULL,
+                            NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NULLABLE);
+                }
+
+                if (isMethodOverridden) {
+                    checkContainsAny(MSG_NEED_TO_INHERIT_PARAM_ANNOTATIONS,
+                            NullnessAnnotation.PARAMETERS_ARE_NONNULL_BY_DEFAULT);
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Class ConstructorJsr305Check. Check a constructor.
+     */
+    private final class ConstructorJsr305Check extends AbstractMethodJsr305Check {
```

Comment:
If would be good to not use Check in names for classes other than top class

Reply:
Done with next commit.